lucene/core/src/java/org/apache/lucene/codecs/lucene90/Lucene90VectorReader.java

```diff
@@ -386,9 +387,18 @@ public int nextDoc() {
     }
 
     @Override
-    public int advance(int target) throws IOException {
-      // We could do better by log-binary search in ordToDoc, but this is never used
-      return slowAdvance(target);
+    public int advance(int target) {
+      assert docID() < target;
+      ord = Arrays.binarySearch(fieldEntry.ordToDoc, target);
```

Comment:
I should have used `Arrays.binarySearch(fieldEntry.ordToDoc,  ord + 1,  fieldEntry.ordToDoc.length - 1,  target)`. Will update